### ISSUE TITLE
JSON file being created when using the UseEntityFrameworkProvider

### ISSUE THREAD

**Issue by mrlife** [thepirat000/Audit.NET#379]
**Describe the bug**
Everything seems to work fine with the `UseEntityFrameworkProvider` Data Provider, however, a `.json` file is created for the first call to `SaveChanges()`, even if that first call to `SaveChanges()` doesn't include an audited entity.

That, and I noticed the following in the docs, which makes me think this may be a bug, since I am using a Data Provider:

> If you don't specify a Data Provider, a default FileDataProvider will be used to write the events as .json files into the current working directory.

If I misunderstood, and this is a feature, I haven't been able to find how to turn it off.

I am configuring like the following (except with my own classes):

```
Audit.Core.Configuration.Setup()
    .UseEntityFramework(ef => ef
        .AuditTypeExplicitMapper(m => m
            .Map<Order, Audit_Order>()
            .Map<OrderItem, Audit_OrderItem>()
            .AuditEntityAction<IAudit>((evt, entry, auditEntity) =>
            {
                auditEntity.AuditDate = DateTime.UtcNow;
                auditEntity.UserName = evt.Environment.UserName;
                auditEntity.AuditAction = entry.Action; // Insert, Update, Delete
            })
        )
    );
```

**To Reproduce**
Start the project and notice the `.json` file is created.

**Expected behavior**
A `.json` file is not created when using the `UseEntityFrameworkProvider` Data Provider.

**Libraries (specify the Audit.NET extensions being used including version):**
 - Audit.EntityFramework.Core: 16.5.4

**Target .NET framework:**
 - .NET 5

**Comment by thepirat000** [comment]
What's the JSON file contents?

You should call the configuration `Audit.Core.Configuration.Setup().UseEntityFramework()` on your Startup or Program.Main. If you do it on the static constructor of the context, any audit event triggered before accesing the dbcontext for the first time will use the default data provider which is to create the JSON file.

Maybe the seeding of the database if causing the audit event to trigger? 

You have many options, but you should avoid having the setup on static constructors whenever you can, and put it on your application startup code.

You can also temporarily disable the audits globally with `Audit.Core.Configuration.AuditDisabled = true;` and then re-enable it by setting the config to `false`

**Comment by mrlife** [comment]
Since I am using ASP.NET Core Blazor (server-side), I won't have access to the user's login name from Startup or Program.Main due to services being scoped to the Blazor circuit/workflow. See [this comment](https://github.com/dotnet/aspnetcore/issues/30982#issuecomment-801316976) on an issue I created on the ASP.NET Core project.

Perhaps calling `Audit.Core.Configuration.AuditDisabled = true;` in Startup.Configure or Program.Main (is one preferable over the other?) and then calling `Audit.Core.Configuration.AuditDisabled = false;` on the line before I call `Audit.Core.Configuration.Setup()` would be good?

**Comment by thepirat000** [comment]
Haven't used blazor yet, but just for curiosity, how do you access the login name on the dbcontext static constructor then?. 

Setting AuditDisable will do the trick, but will be more a workaround if you say that something that should not be audited is actually being audited...

Can you share a sample project that reproduces the issues?

**Comment by mrlife** [comment]
TL;DR - I call `Audit.Core.Configuration.Setup()` and have access to Identity claims for the user, and can get the login name, in `App.razor`'s `OnInitialized()` lifecycle event.

More detailed answer:

Blazor server-side has one _**razor page**_ called `_Host.cshtml`, which is called automatically when a user accesses any URL of the application. This razor page calls the top-most _**razor component**_ called `App.razor`.

Since the login name can only be accessed from Identity within the Blazor workflow (i.e. in razor components), I call `Audit.Core.Configuration.Setup()` when `App.razor` is being initialized (in its `OnInitialized()` lifecycle event that can be overridden to provide custom logic).

---

Setting `AuditDisabled` seems to work fine for this situation. Thank you for the solution!

This issue is solved for me, but here is a [sample app](https://github.com/thepirat000/Audit.NET/files/6160202/BlazorServerDbContextExample.zip) to check out the issue. Note that the `.json` files are created only when `ContactsDb.db` is created/seeded, as you suspected earlier in this issue, so it's not exactly the same issue as mine, since there's no migration needed for the built-in SQLite database and seeding is done on first app run. You'll have to delete `ContactsDb.db` each time you want to replicate the issue.
